web/wp-content/themes/bumblebee/inc/menu-walker-tagging.php

```diff
@@ -0,0 +1,320 @@
+		$json_attributes  = json_encode( $attributes_array );
+		$attributes      .= ' data-analytics-metrics=\''.$json_attributes.'\'';
+		$attributes      .= ! empty( $item->url ) ? ' href="' . esc_attr( $item->url ) .'"' : '';
+
+		$item_output = $args->before;
+		$item_output .= '<a'. $attributes .'>';
+		$item_output .= $args->link_before . apply_filters( 'the_title', $item->title, $item->ID ) . $args->link_after;
+		$item_output .= '</a>';
+		$item_output .= $args->after;
+
+		$output .= apply_filters( 'walker_nav_menu_start_el', $item_output, $item, $depth, $args );
+	}
+
+	function get_social_profile_names( $title ) {
+		if ( strpos( $title, 'facebook' ) !== false ) {
+			return( 'facebook' );
+		}
+		if ( strpos( $title, 'pinterest' ) !== false ) {
+			return( 'pinterest' );
+		}
+		if ( strpos( $title, 'twitter' ) !== false ) {
+			return( 'twitter' );
+		}
+		if ( strpos( $title, 'google-plus' ) !== false ) {
+			return( 'google plus' );
+		}
+		if ( strpos( $title, 'instagram' ) !== false ) {
+			return( 'instagram' );
+		}
+		if ( strpos( $title, 'youtube' ) !== false ) {
+			return( 'youtube' );
+		}
+		if ( strpos( $title, 'envelope' ) !== false ) {
+			return( 'newsletter signup' );
+		}
+		return( $title );
+	}
+}
+
+/*
+ * Adding Additional analytic attribute to Footer links
+ */
+class V2_Footer_Links extends Walker_Nav_Menu {
+	function start_el( &$output, $item, $depth = 0, $args = array(), $id = 0 ) {
+		global $wp_query;
+		$indent = ( $depth ) ? str_repeat( "\t", $depth ) : '';
+		$class_names = $value = '';
+		$classes = empty( $item->classes ) ? array() : (array) $item->classes;
+		$class_names = join( ' ', apply_filters( 'nav_menu_css_class', array_filter( $classes ), $item ) );
+		$class_names = ' class="' . esc_attr( $class_names ) . '"';
+		$output .= $indent . '<li id="menu-item-'. $item->ID . '"' . $value . $class_names .'>';
+
+		$attributes = ! empty( $item->attr_title ) ? ' title="' . esc_attr( $item->attr_title ) .'"' : '';
+		$attributes .= ! empty( $item->target ) ? ' target="' . esc_attr( $item->target ) .'"' : '';
+		$attributes .= ! empty( $item->xfn ) ? ' rel="' . esc_attr( $item->xfn ) .'"' : '';
+
+		$attributes_array = array( 'link_name' => esc_attr( $item->title ),'link_module' => 'navigation','link_pos' => 'footer' );
+		$json_attributes = json_encode( $attributes_array );
+		$attributes .= ' data-analytics-metrics=\''.$json_attributes.'\'';
+		$attributes .= ! empty( $item->url ) ? ' href="' . esc_attr( $item->url ) .'"' : '';
+
+		$item_output = $args->before;
+		$item_output .= '<a'. $attributes .'>';
+		$item_output .= $args->link_before . apply_filters( 'the_title', $item->title, $item->ID ) . $args->link_after;
+		$item_output .= '</a>';
+		$item_output .= $args->after;
+
+		$output .= apply_filters( 'walker_nav_menu_start_el', $item_output, $item, $depth, $args );
+	}
+}
+
+/*
+ * Adding Additional analytic attribute to Footer links
+ */
+class Menu_Links extends Walker_Nav_Menu {
```

Comment:
This class is identical to to `V2_Footer_Links`, except for its name and `$attributes_array` variable. `Logged_In_Menu` seems pretty similar too. And `TMBI_Social_Profiles` too, besides the extra function and white space differences.
Classes that similar can probably be rewritten in a single class with appropriate extension points handling the common code, and subclasses containing the differences _only_.

Reply:
See https://readersdigest.atlassian.net/browse/DRDT-103